core-bundle/src/Resources/contao/modules/ModuleLogin.php

```diff
@@ -69,14 +70,42 @@ protected function compile()
 	{
 		$container = System::getContainer();
 
-		/** @var RouterInterface $router */
+		/** @var Router $router */
 		$router = $container->get('router');
 
-		if ($container->get('contao.security.token_checker')->hasFrontendUser())
+		/** @var TokenChecker $tokenChecker */
+		$tokenChecker = $container->get('contao.security.token_checker');
+
+		/** @var AuthenticationException|null $exception */
+		$exception = $container->get('security.authentication_utils')->getLastAuthenticationError();
+
+		/** @var PageModel $objPage */
+		global $objPage;
+
+		if ($tokenChecker->hasTwoFactorToken())
 		{
-			/** @var PageModel $objPage */
-			global $objPage;
+			$user = FrontendUser::getInstance();
+			$redirectPage = PageModel::findByPk($this->jumpTo);
```

Comment:
`$redirectPage = $this->jumpTo > 0 ? PageModel::findByPk($this->jumpTo) : null;`

Reply:
Fixed in 9a31a6c57.